### PR TITLE
Update to latest Elixir and OAuth2, fix issue with get_token!

### LINKED ISSUE #1: Unable to complete OAuth token exchange portion of authorization
Running into an issue with the token exchange portion of authenticating. Error as follows

```key :access_token not found in: %OAuth2.Client{authorize_url: "https://www.strava.com/oauth/authorize", client_id: "<<client_id>>", client_secret: "<<secret>>", headers: [], params: %{}, redirect_uri: "http://localhost:4000/auth/strava/callback", site: "https://www.strava.com/", strategy: Ueberauth.Strategy.Strava.OAuth, token: %OAuth2.AccessToken{access_token: nil, expires_at: nil, other_params: %{"errors" => [%{"code" => "invalid", "field" => "", "resource" => "Application"}], "message" => "Authorization Error"}, refresh_token: nil, token_type: "Bearer"}, token_method: :post, token_url: "https://www.strava.com/oauth/token"}```

I've verified that the code is in fact returned from the authorization route, so there's something weird with the hand off to the token route

### PR DESCRIPTION
**Pull request by vinniefranco**
* Updates to latest version of Ueberauth
* Changes things a bit to go off latest github example
* Updates `get_token!` to explicitly pass `client_id` and `client_secret` which fixes #1 
* Verified with latest example app

### CODE REVIEW

File: lib/ueberauth/strategy/strava.ex [hunk 1]
```diff
@@ -13,22 +15,26 @@ defmodule Ueberauth.Strategy.Strava do
   Handles initial request for Strava authentication.
   """
   def handle_request!(conn) do
+    scopes = conn.params["scope"] || option(conn, :default_scope)
+    opts = [redirect_uri: callback_url(conn), scope: scopes]
 
-    authorize_url = conn.params
```
Comment (andrewhao) [inline lib/ueberauth/strategy/strava.ex:17]: Any reason why this block was taken out?

File: lib/ueberauth/strategy/strava.ex [hunk 2]
```diff
@@ -13,22 +15,26 @@ defmodule Ueberauth.Strategy.Strava do
...
-      |> maybe_replace_param(conn, "scope", :default_scope)
-      |> Enum.map(fn {k,v} -> {String.to_existing_atom(k), v} end)
-      |> Keyword.put(:redirect_uri, callback_url(conn))
-      |> Ueberauth.Strategy.Strava.OAuth.authorize_url!
+    opts = if conn.params["state"] do 
+             Keyword.put(opts, :state, conn.params["state"])
```
Comment (andrewhao) [inline lib/ueberauth/strategy/strava.ex:22]: I think L19 in the old code accomplished this similarly:

```elixir
|> Enum.map(fn {k,v} -> {String.to_existing_atom(k), v} end)
```

**Review by andrewhao — COMMENTED** [review]
I verified this was working with another app - thanks for the fixes! I do have some general questions about the approach - particularly around why the variables need to be manually merged in (I'm looking at the source for the `ueberauth_facebook` project and noticing that they don't manually merge the `client_id` param on their side (https://github.com/ueberauth/ueberauth_facebook/blob/master/lib/ueberauth/strategy/facebook.ex). It could be worth investigating whether there was a breaking change in the OAuth library implementation that might have caused this issue.

In any case, you can run forward with this as much as you want, or leave this as-is. If this is as far as you want to take this, I'll merge and refactor from here.

File: lib/ueberauth/strategy/strava.ex (hunk 1 above)
Comment (vinnie-eb) [reply lib/ueberauth/strategy/strava.ex:17]: I went with the approach that the more recent strategies use: https://github.com/ueberauth/ueberauth_github/blob/master/lib/ueberauth/strategy/github.ex#L89

File: lib/ueberauth/strategy/strava.ex (hunk 2 above)
Comment (vinnie-eb) [reply lib/ueberauth/strategy/strava.ex:22]: Again, just went with more recent strategies, I can revert if it doesn't suite taste

**Comment by vinnie-eb** [comment]
> I verified this was working with another app - thanks for the fixes! I do have some general questions about the approach - particularly around why the variables need to be manually merged in (I'm looking at the source for the ueberauth_facebook project and noticing that they don't manually merge the client_id param on their side 

I noticed that as well. This PR updates OAuth2 to the same release as the FB strategy that you provided... so not too sure there. I just know that it did not work until I explicitly provided `client_secret`, and `client_id`

**Comment by andrewhao** [comment]
@vinnie-eb - your work here makes sense. I was comparing it to the code in the ueberauth_facebook repo, which takes yet a different approach. In any case, I'll add some tests, which is the biggest missing piece, anyways.

**Pull request merged**